docs/introduction/asynchronous-tasks-with-rabbitmq.md

```diff
@@ -0,0 +1,151 @@
+
+```diff
+product-visibility-recalculate-consumer:
+    <<: *common_consumer_configuration
+    container_name: shopsys-framework-product-visibility-recalculate-consumer
+    command: product_visibility_recalculate
++   environment:
++       CONSUMER_RUNS_BEFORE_RESTART: 2048
+```
+
+_Note: default number of processed messages is 1000_
+
+### In Kubernetes
+
+In Kubernetes, consumers are, by default, placed into the php-fpm pod as another container.
+This is to give the consumer access to the [application filesystem](./abstract-filesystem.md).
+
+One of consumer's container configurations, placed into `webserver-php-fpm` deployment as the another container, look like this:
+
+```yaml
+-   image: ~
+    name: product-visibility-recalculate-consumer
+    securityContext:
+        runAsUser: 33
+    workingDir: /var/www/html
+    command:
+        - docker-php-consumer-entrypoint
+    args:
+        - product_visibility_recalculate
+    volumeMounts:
+        -   name: source-codes
+            mountPath: /var/www/html
+        -   name: domains-urls
+            mountPath: /var/www/html/app/config/domains_urls.yml
+            subPath: domains_urls.yml
+        -   name: parameters
+            mountPath: /var/www/html/app/config/parameters.yml
+            subPath: parameters.yml
+    env:
+        -   name: GOOGLE_CLOUD_STORAGE_BUCKET_NAME
+            value: ~
+        -   name: GOOGLE_CLOUD_PROJECT_ID
+            value: ~
+```
+
+_Note: You should keep php-fpm container as the first (therefore the default one) in container list in the deployment_
+
+Most interesting is `args` key, which represents what consumer should be run within this container and this name is the same as is configured in `old_sound_rabbit_mq.yml` file.
+
+If you want to run another consumer, you just need to add another container with the same configuration as above and specify proper consumer in `args` field.
+
+Other placeholder values, represented by `~`, should be updated by your build script with proper ones for this container too, so don't forget to check your build tools.
+
+If you need to change the number of messages processed before the consumer is restarted, you can adjust the environment variable `CONSUMER_RUNS_BEFORE_RESTART`
+
+```diff
+    env:
+        -   name: GOOGLE_CLOUD_STORAGE_BUCKET_NAME
+            value: ~
+        -   name: GOOGLE_CLOUD_PROJECT_ID
+            value: ~
++       -   name: CONSUMER_RUNS_BEFORE_RESTART
++           value: 2048
+```
+
+_Note: default number of processed messages is 1000_
+
+### Native installation / production
+
+For natively installed application we recommend to use [Supervisor](http://supervisord.org) to ensure your consumer is up and running.
+
+If you want to run consumers with Supervisor or add your own, please take a look at the [Native installation guide](../installation/native-installation.md#run-background-processing-with-supervisor) where you find the instructions.
+
+## Develop a consumer
```

Comment:
```suggestion
## Consumers in development environment
```

Your title makes me feel that I will learn how to develop my own consumer, but it is not true